packages/component-base/test/resizable-mixin.test.js

```diff
@@ -0,0 +1,57 @@
+import { expect } from '@esm-bundle/chai';
+import { fixtureSync } from '@vaadin/testing-helpers';
+import sinon from 'sinon';
+import { html, PolymerElement } from '@polymer/polymer/polymer-element.js';
+import { ResizableMixin } from '../src/resizable-mixin.js';
+
+describe('resizable-mixin', () => {
+  let element;
+
+  before(() => {
+    customElements.define(
+      'resizable-mixin-element',
+      class extends ResizableMixin(PolymerElement) {
+        static get template() {
+          return html`
+            <style>
+              :host {
+                display: block;
+              }
+            </style>
+            <div></div>
+          `;
+        }
+
+        _onResize() {
+          this.__resolveResize?.();
+        }
+
+        nextResize() {
+          return new Promise((resolve) => {
+            this.__resolveResize = resolve;
+          });
+        }
+      }
+    );
+  });
+
+  beforeEach(async () => {
+    element = fixtureSync(`<resizable-mixin-element></resizable-mixin-element>`);
+    // Wait for the initial resize.
+    await element.nextResize();
+  });
+
+  it('should notify resize', async () => {
+    element.style.width = '100px';
+    await element.nextResize();
```

Comment:
Using await here would just make the test stall until timeout. The WICG test suite seems to use a specific timeout instead: https://github.com/WICG/resize-observer/blob/master/test/resources/resizeTestHelper.js

But with the current amount of tests the current solution is probably good enough.

Reply:
Yeh, an arbitrary timeout doesn't sound tempting either and it slows down the test run

Reply:
Alright